python/ray/autoscaler/_private/monitor.py

```diff
@@ -177,7 +228,35 @@ def update_load_metrics(self):
             request, timeout=4)
         resources_batch_data = response.resource_usage_data
 
+        # Tell the readonly node provider what nodes to report.
+        if self.readonly_config:
+            new_nodes = []
+            for msg in list(resources_batch_data.batch):
+                node_id = msg.node_id.hex()
+                new_nodes.append((node_id, msg.node_manager_address))
+            self.autoscaler.provider._set_nodes(new_nodes)
+
+        mirror_node_types = {}
+        cluster_full = False
         for resource_message in resources_batch_data.batch:
+            # Generate node type config based on GCS reported node list.
+            if self.readonly_config:
+                # Keep prefix in sync with ReadonlyNodeProvider.
+                node_type = format_readonly_node_type(
+                    resource_message.node_id.hex())
+                resources = resource_message.resources_total
+                for k in list(resources.keys()):
+                    if k.startswith("node:"):
+                        del resources[k]
+                mirror_node_types[node_type] = {
```

Comment:
Do we run into any issues with temporarily missing resource_messages for certain node_ids?
I have in mind "missing heartbeats" heartbeats situations but don't know the details of how resources_batch_data is populated.